### PR TITLE
fix: handle devices by their db entity id

### PR DESCRIPTION
**Pull request by st3iny**
Fix https://github.com/nextcloud/twofactor_webauthn/issues/176
Fix https://github.com/nextcloud/twofactor_webauthn/issues/281

Devices should be identified by their entity ids when toggling or removing them. The public key credential id is not unique.

### CODE REVIEW

File: lib/Db/PublicKeyCredentialEntityMapper.php [hunk 1]
```diff
@@ -93,6 +93,23 @@ public function __construct(IDBConnection $db) {
...
+		$qb->select('*')
+			->from($this->getTableName())
+			->where($qb->expr()->eq(
+				'id',
+				$qb->createNamedParameter($id, IQueryBuilder::PARAM_INT),
+				IQueryBuilder::PARAM_INT,
+			));
+		try {
+			return $this->findEntity($qb);
+		} catch (\Exception) {
```
Comment (ChristophWurst) [inline lib/Db/PublicKeyCredentialEntityMapper.php:108]: a bit too generic for my taste :speak_no_evil:

Comment (ChristophWurst) [reply lib/Db/PublicKeyCredentialEntityMapper.php:108]: better: catch not found exception only

Comment (st3iny) [reply lib/Db/PublicKeyCredentialEntityMapper.php]: Done

**Review by ChristophWurst — APPROVED** [review]

File: tests/Unit/Service/WebAuthnManagerTest.php [hunk 2]
```diff
@@ -110,10 +110,11 @@ public function testGetNoDevices(): void {
...
 	public function testDisableWebAuthn(): void {
 		$user = $this->createMock(IUser::class);
-		$reg = $this->createMock(PublicKeyCredentialEntity::class);
+		$reg = new PublicKeyCredentialEntity();
+		$reg->setId(420);
 		$this->mapper->expects(self::once())
-			->method('findPublicKeyCredential')
-			->with('k1')
+			->method('findById')
+			->with(420)
```
Comment (ChristophWurst) [inline tests/Unit/Service/WebAuthnManagerTest.php:117]: 🥦

**Review by ChristophWurst — APPROVED** [review]

File: lib/Db/PublicKeyCredentialEntityMapper.php [hunk 3]
```diff
@@ -93,6 +94,26 @@ public function __construct(IDBConnection $db) {
...
+		$qb->select('*')
+			->from($this->getTableName())
+			->where($qb->expr()->eq(
+				'id',
+				$qb->createNamedParameter($id, IQueryBuilder::PARAM_INT),
+				IQueryBuilder::PARAM_INT
+			));
+		try {
+			return $this->findEntity($qb);
+		} catch (DoesNotExistException) {
```
Comment (ChristophWurst) [inline lib/Db/PublicKeyCredentialEntityMapper.php:112]:
```suggestion
		} catch (DoesNotExistException $e) {
```

for php7.4 compat

Comment (st3iny) [reply lib/Db/PublicKeyCredentialEntityMapper.php]: Done

**Pull request merged**